cluster/src/main/java/com/hartwig/pipeline/metadata/AddDatatypeToFile.java

```diff
@@ -14,10 +14,11 @@
     private final DataType datatype;
     private final String barcode;
 
-    public AddDatatypeToFile(DataType datatype, Folder folder, String namespace, String filename, final String barcode) {
+    public AddDatatypeToFile(final DataType datatype, final Folder folder, final String namespace, final String filename,
+            final String barcode) {
         this.datatype = datatype;
         this.barcode = barcode;
-        String namespacedFile = namespace + "/" + filename;
+        String namespacedFile = filename.isEmpty() ? namespace : namespace + "/" + filename;
```

Comment:
i think an `isDir` + a non-empty filename will be easier to maintain in the long run (even if the filename and the namespace are the same)